hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/RatisBlockOutputStream.java

```diff
@@ -126,4 +128,16 @@ void waitOnFlushFutures() throws InterruptedException, ExecutionException {
   void cleanup() {
     commitWatcher.cleanup();
   }
+
+  @Override
+  public void hflush() throws IOException {
+    hsync();
+  }
+
+  @Override
+  public void hsync() throws IOException {
+    if (!isClosed()) {
```

Comment:
watchForCommit is basically to make sure buffers are properly send and stored at DNs. 
SO, here is the place we need to implement to notify to OM that, here is a hsync call from app, we need to update the block info along with the so far committed content right? If, yes, do you mind adding a todo comment here to make it more clear? If we have already filed followup JIRA for om handling, may be a good idea to put in the comment to make it more clear if anybody wants to review.

Reply:
> ... here is the place we need to implement to notify to OM, here is a hsync call from app, we need to update the block info along with the so far committed content right? ...

This is a lower level method (the commit here is Ratis commit, not OM commit).

Apps use OzoneOutputStream (or OzoneFSOutputStream).  OzoneOutputStream contains KeyOutputStream (which could be wrapped by a CryptoOutputStream).  The right place to update OM length is `KeyOutputStream.hsync()`.  Just have filed HDDS-7724 and revised the TODO in `KeyOutputStream.hsync`.